runtime/src/dispatcher.rs

```diff
@@ -253,6 +278,10 @@ impl Dispatcher {
                         signed_policy_raw,
                     );
                 }
+                Ok((_ctx, _id, Body::RuntimeAbortRequest {})) => {
+                    info!(self.logger, "Received abort request");
```

Comment:
Add a comment explaining that if this "abort request" is processed here this means that the abort has already been processed.

Reply:
On another thought, since you have this `abort_and_wait` why do you need to process the request here at all? Maybe just at the beginning of the loop (e.g., before `rx.recv()`) check if the abort flag is set and if so, clear it and emit something to `abort_tx`? This seems cleaner than having this strange request here.